### ISSUE TITLE
False positives in org.openrewrite.staticanalysis.SimplifyBooleanExpression.

### ISSUE THREAD

**Issue by motlin** [openrewrite/rewrite-static-analysis#423]
## What version of OpenRewrite are you using?

I am trying to accept a dependabot upgrade of rewrite-maven-plugin from 5.46.3 to 5.47.2. This pulls in an upgrade of the rule org.openrewrite.staticanalysis.SimplifyBooleanExpression which attempts to make new modifications to my source code.

## What is the smallest, simplest way to reproduce the problem?

Clone the [liftwizard project](https://github.com/liftwizard/liftwizard) and run `just rewrite org.openrewrite.staticanalysis.SimplifyBooleanExpression`

## What did you expect to see?

No change.

## What did you see instead?

Three modifications. All changed negated equals conditions in if-statements to the literal `true`. One example, removing irrelevant code is:

```java
    @Override
    protected void assertFileContentsOrThrow(
            @Nonnull String resourceClassPathLocation,
            @Nonnull String actualString)
            throws URISyntaxException, IOException
    {
        InputStream inputStream = this.callingClass.getResourceAsStream(resourceClassPathLocation);
        Objects.requireNonNull(inputStream, () -> resourceClassPathLocation + " not found.");
        String expectedStringFromFile = slurp(inputStream, StandardCharsets.UTF_8);

        if (!actualString.equals(expectedStringFromFile))
        {
            // ...
        }
    }
```

Got replace with `if (true)`.

**Comment by timtebeek** [comment]
Hi @motlin ; thanks for the report! This has since been fixed and we'll release a new version patch of OpenRewrite & plugins:
- https://github.com/openrewrite/rewrite/commit/9cfddab7dee4de929376839061b22c4e9a363de2
- https://github.com/openrewrite/rewrite/issues/4821
